tests/src/test/java/org/eclipse/hono/tests/client/HonoKafkaConsumerIT.java

```diff
@@ -124,9 +126,10 @@ void printTestInfo(final TestInfo testInfo) {
      */
     @AfterEach
     void closeConsumer(final VertxTestContext ctx) {
-        if (kafkaConsumer != null) {
-            kafkaConsumer.stop().onComplete(ctx.succeedingThenComplete());
-        }
+        Future.all(
+                Optional.ofNullable(kafkaConsumer).map(k -> k.stop()).orElseGet(() -> Future.succeededFuture()),
+                Optional.ofNullable(kafkaProducer).map(p -> p.close()).orElseGet(() -> Future.succeededFuture()))
+            .onComplete(ctx.succeedingThenComplete());
```

Comment:
Now that also the producer gets closed, the method name and javadoc should be adapted.